src/cpp/src/whisper/whisper.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <iostream>
+#include <openvino/openvino.hpp>
+#include <regex>
+#include <thread>
+
+#include "../utils.hpp"
+#include "openvino/genai/streamer_base.hpp"
+#include "openvino/genai/whisper_generation_config.hpp"
+#include "openvino/genai/whisper_speech_recognition_pipeline.hpp"
+#include "whisper_feature_extractor.hpp"
+#include "whisper_models.hpp"
+
+namespace {
+
+void suppress_tokens(ov::Tensor& logits, const std::vector<int64_t>& suppress_tokens) {
+    auto data = logits.data<float>();
+    for (auto supress_token : suppress_tokens) {
+        data[supress_token] = -std::numeric_limits<float>::infinity();
+    }
+}
+
+ov::Tensor encode(ov::InferRequest& request, std::vector<float>& mel_data) {
+    ov::Shape input_shape = {1, 80, 3000};
```

Comment:
is it possible to read from some configs instead of hardcoded values?

Reply:
I reused input tensor shape dim values instead of hardcoded ones:

```c++
    auto shape = request.get_tensor("input_features").get_shape();

    OPENVINO_ASSERT(mel_data.size() == shape[1] * shape[2],
                    "Mel spectrogram required size: ",
                    shape[1],
                    " * ",
                    shape[2],
                    ". Actual size: ",
                    mel_data.size(),
                    ".");

    ov::Tensor input_tensor(ov::element::f32, {1, shape[1], shape[2]}, mel_data.data());
```